sympy/integrals/heurisch.py

```diff
@@ -558,8 +558,20 @@ def find_non_syms(expr):
         if solution is None:
             return None
         else:
-            solution = [ (k.as_expr(), v.as_expr()) for k, v in solution.items() ]
-            return candidate.subs(solution).subs(list(zip(coeffs, [S.Zero]*len(coeffs))))
+            sol = {}
+            # remove coeff from k in case Floats were involved
```

Comment:
With AlgebraicField the situation is even worse. Its dtype is ANP which has no attribute 'as_expr'.